jobbergate-agent/jobbergate_agent/utils/user_mapper.py

```diff
@@ -0,0 +1,68 @@
+"""
+Provide to the agent a way to map email addresses from Jobbergate local Slurm users.
+
+Custom mappers can be added to the agent as installable plugins, which are discovered at runtime.
+"""
+from dataclasses import dataclass
+from typing import Protocol
+
+from buzz import enforce_defined, require_condition
+
+from jobbergate_agent.settings import SETTINGS
+from jobbergate_agent.utils.logging import logger
+from jobbergate_agent.utils.plugin import load_plugins
+
+
+class SlurmUserMapper(Protocol):
```

Comment:
I was in doubt about this one, wondering if `collections.Mapping` would be a better choice since it is in fact a map. It would allow dictionaries and many others to work just out of the box. An easy change still while in pre-release. The downside is that `SingleUserMapper` would look complicated.